Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2160,14 +2160,14 @@ void WebProcessProxy::didExceedCPULimit()
 
 #if PLATFORM(MAC) && USE(RUNNINGBOARD)
     // This background WebProcess is using too much CPU so we try to suspend it if possible.
-    if (runningBoardThrottlingEnabled() && !throttler().isSuspended()) {
+    if (runningBoardThrottlingEnabled() && !throttler().isSuspended() && !isRunningServiceWorkers()) {
```

Comment:
I was just looking at this code snippet last week and was wondering how it would make sense if the process was hosting a worker...